feature-libs/product-configurator/rulebased/core/state/effects/configurator-basic-effect.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { Configurator } from '../../model/configurator.model';
+
+/**
+ * Service that provides helper methods for the basic configurator effects,
+ * in order to enhance them without the need to introduce new effects
+ */
+@Injectable({ providedIn: 'root' })
+export class ConfiguratorBasicEffectService {
+  /**
+   * Finds first group with attributes for a configuration. Throws error if such a group does not exist,
+   * as this is an illegal state
+   * @param configuration
+   * @returns Group id
+   */
+  getGroupWithAttributesForConfiguration(
+    configuration: Configurator.Configuration
+  ): string {
+    const id = this.getGroupWithAttributes(configuration.groups);
+    if (id) {
+      return id;
+    } else {
+      throw new Error('Configuration does not have any attributes');
+    }
+  }
+  /**
+   * Finds first group with attributes in a list of groups
+   * @param groups
+   * @returns Group or undefined if such a group does not exist
+   */
+  getGroupWithAttributes(groups: Configurator.Group[]): string | undefined {
```

Comment:
What about to make this method `protected`?

Reply:
Yes, good point